Market proposal goes straight into continuous/enacted rather than opening auction and pending.

# Problem encountered
When you submit a proposal for a new market, which passes, it goes straight into "Continuous" and "Active", rather than "Opening Auction" and "Pending". The underlying proposal still shows the correct state, and orders on the book do not cross, so this may just be a reported state (data node?) problem.

# Observed behaviour
New market reports state as "Continuous" and "Active"

# Expected behaviour
New market reports state as "Opening Auction" and "Pending"

# System response
There is a WIP checkpoint test which triggers this. It can be found on [this branch](https://github.com/vegaprotocol/system-tests/commit/c1b3628710d150403a38bf78ca8fa7b1086d7f23). At time of writing, this test does not actually do a checkpoint restore, and then checks the state of the market, which is not as expected. Run the test with this command from the top-level ST dir:
```pytest -k test_accepted_non_enacted_market_restored```
```
>       assert market_after["state"] == "STATE_PENDING"
E       AssertionError: assert 'STATE_ACTIVE' == 'STATE_PENDING'
E         - STATE_PENDING
E         + STATE_ACTIVE
```

# Steps to reproduce
- Propose a new market with enactment time 3 days into the future (Without the delayed enactment date, you still see the market enter the wrong trading mode state, but the "Pending"/"Active" issue requires a market which is not yet enacted)
- Pass the vote.
- Submit two orders which could cross, and another so that the market would remain open (Proves that the market is actually still in the auction period, since these orders do not cross)

# Evidence
```
{
  "data": {
    "proposal": {
      "id": "538494c750bd4cf23bcb31613400cc270092688db3da759ade22b1ae0508194c",
      "state": "Passed",
      "datetime": "2022-04-06T13:57:28.3291792Z",
      "terms": {
        "closingDatetime": "2022-04-06T13:57:37Z",
        "enactmentDatetime": "2022-04-09T13:57:29Z"
      }
    },
    "market": {
      "id": "538494c750bd4cf23bcb31613400cc270092688db3da759ade22b1ae0508194c",
      "state": "Active",
      "proposal": {
        "id": "538494c750bd4cf23bcb31613400cc270092688db3da759ade22b1ae0508194c",
        "state": "Passed"
      },
      "tradingMode": "Continuous",
      "orders": [
        {
          "status": "Active",
          "price": "950",
          "side": "Buy",
          "party": {
            "id": "5f0101ea1da0102ff5569462da7b697f0a1c2c1a2cadb1aa1a502760c5603a13",
            "accounts": [
              {
                "asset": {
                  "id": "1a4684ab6883064b70a365cfd1a5320518b1fd1af8929660b002dea37698f45d"
                },
                "balance": "99999880"
              },
              {
                "asset": {
                  "id": "1a4684ab6883064b70a365cfd1a5320518b1fd1af8929660b002dea37698f45d"
                },
                "balance": "120"
              }
            ]
          },
          "rejectionReason": null
        },
        {
          "status": "Active",
          "price": "950",
          "side": "Sell",
          "party": {
            "id": "231e1b85d858c1fa7833178f874393504e06f74cd5a9ea855dff5591a81a415d",
            "accounts": [
              {
                "asset": {
                  "id": "1a4684ab6883064b70a365cfd1a5320518b1fd1af8929660b002dea37698f45d"
                },
                "balance": "99999874"
              },
              {
                "asset": {
                  "id": "1a4684ab6883064b70a365cfd1a5320518b1fd1af8929660b002dea37698f45d"
                },
                "balance": "126"
              }
            ]
          },
          "rejectionReason": null
        },
        {
          "status": "Active",
          "price": "950",
          "side": "Sell",
          "party": {
            "id": "6022c8f068e4476a87985f2908fed99db33349b3d9403ab3fe4f9083fe6b2440",
            "accounts": [
              {
                "asset": {
                  "id": "1a4684ab6883064b70a365cfd1a5320518b1fd1af8929660b002dea37698f45d"
                },
                "balance": "99999874"
              },
              {
                "asset": {
                  "id": "1a4684ab6883064b70a365cfd1a5320518b1fd1af8929660b002dea37698f45d"
                },
                "balance": "126"
              }
            ]
          },
          "rejectionReason": null
        }
      ]
    }
  }
}
```

# Definition of Done
>ℹ️ Not every issue will need every item checked, however, every item on this list should be properly considered and actioned to meet the [DoD](https://github.com/vegaprotocol/vega/blob/develop/DEFINITION_OF_DONE.md).

**Before Merging**
- [ ] Code refactored to meet SOLID and other code design principles
- [ ] Code is compilation error, warning, and hint free
- [ ] Carry out a basic happy path end-to-end check of the new code
- [ ] All APIs are documented so auto-generated documentation is created
- [ ] All bug recreation steps can be followed without presenting the original error/bug
- [ ] All Unit, Integration and BVT tests are passing
- [ ] Implementation is peer reviewed (coding standards, meeting acceptance criteria, code/design quality)
- [ ] Create [front end](https://github.com/vegaprotocol/token-frontend/issues) or [console](https://github.com/vegaprotocol/console/issues) tickets with feature labels (should be done when starting the work if dependencies known i.e. API changes)

**After Merging**
- [ ] Move development ticket to `Done` if there is **NO** requirement for new [system-tests](https://github.com/vegaprotocol/system-tests/issues)
- [ ] Resolve any issues with broken system-tests
- [ ] Create [documentation](https://github.com/vegaprotocol/documentation/issues) tickets with feature labels if functionality has changed, or is a new feature

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -21,6 +21,7 @@
 - [5520](https://github.com/vegaprotocol/vega/issues/5520) - Do not fail silently when wallet fails to start
 - [5521](https://github.com/vegaprotocol/vega/issues/5521) - Fix asset bundle and add asset status
 - [5546](https://github.com/vegaprotocol/vega/issues/5546) - Fix collateral checkpoint to unlock locked reward account balance
+- [5194](https://github.com/vegaprotocol/vega/issues/5194) - Fix market trading mode vs market state
 
 ## 0.52.0
 

--- a/execution/market.go
+++ b/execution/market.go
@@ -384,7 +384,7 @@ func NewMarket(
 
 	// The market is initially create in a proposed state
 	mkt.State = types.MarketStateProposed
-	mkt.TradingMode = types.MarketTradingModeContinuous
+	mkt.TradingMode = types.MarketTradingModeNoTrading
 
 	// Populate the market timestamps
 	ts := &types.MarketTimestamps{
@@ -616,6 +616,7 @@ func (m *Market) Reject(ctx context.Context) error {
 	// we closed all parties accounts
 	m.cleanupOnReject(ctx)
 	m.mkt.State = types.MarketStateRejected
+	m.mkt.TradingMode = types.MarketTradingModeNoTrading
 	m.broker.Send(events.NewMarketUpdatedEvent(ctx, *m.mkt))
 	m.stateChanged = true
 
@@ -644,11 +645,13 @@ func (m *Market) StartOpeningAuction(ctx context.Context) error {
 		// we are now in a pending state
 		m.mkt.State = types.MarketStatePending
 		m.mkt.MarketTimestamps.Pending = m.timeService.GetTimeNow().UnixNano()
+		m.mkt.TradingMode = types.MarketTradingModeOpeningAuction
 		m.enterAuction(ctx)
 	} else {
 		// TODO(): to be removed once we don't have market starting
 		// without an opening auction
 		m.mkt.State = types.MarketStateActive
+		m.mkt.TradingMode = types.MarketTradingModeContinuous
 	}
 
 	m.broker.Send(events.NewMarketUpdatedEvent(ctx, *m.mkt))
@@ -805,6 +808,7 @@ func (m *Market) closeMarket(ctx context.Context, t time.Time) error {
 
 	m.broker.Send(events.NewTransferResponse(ctx, clearMarketTransfers))
 	m.mkt.State = types.MarketStateSettled
+	m.mkt.TradingMode = types.MarketTradingModeNoTrading
 	m.broker.Send(events.NewMarketUpdatedEvent(ctx, *m.mkt))
 
 	// remove all order from the book
@@ -930,6 +934,7 @@ func (m *Market) enterAuction(ctx context.Context) {
 
 	if m.as.InAuction() && (m.as.IsLiquidityAuction() || m.as.IsPriceAuction()) {
 		m.mkt.State = types.MarketStateSuspended
+		m.mkt.TradingMode = types.MarketTradingModeMonitoringAuction
 		m.broker.Send(events.NewMarketUpdatedEvent(ctx, *m.mkt))
 		m.stateChanged = true
 	}
@@ -956,6 +961,7 @@ func (m *Market) leaveAuction(ctx context.Context, now time.Time) {
 	defer func() {
 		if !m.as.InAuction() && m.mkt.State == types.MarketStateSuspended {
 			m.mkt.State = types.MarketStateActive
+			m.mkt.TradingMode = types.MarketTradingModeContinuous
 			m.broker.Send(events.NewMarketUpdatedEvent(ctx, *m.mkt))
 			m.stateChanged = true
 		}
@@ -3138,6 +3144,7 @@ func (m *Market) tradingTerminated(ctx context.Context, tt bool) {
 	m.mu.Lock()
 	defer m.mu.Unlock()
 	m.mkt.State = types.MarketStateTradingTerminated
+	m.mkt.TradingMode = types.MarketTradingModeNoTrading
 	m.broker.Send(events.NewMarketUpdatedEvent(ctx, *m.mkt))
 	m.stateChanged = true
 

--- a/go.mod
+++ b/go.mod
@@ -4,7 +4,7 @@ go 1.18
 
 require (
 	code.vegaprotocol.io/oracles-relay v0.0.0-20210201140234-f047e1bf6df3
-	code.vegaprotocol.io/protos v0.52.1-0.20220706074810-352ce32e71c1
+	code.vegaprotocol.io/protos v0.51.2-0.20220613103838-a73da28da1e1
 	code.vegaprotocol.io/quant v0.2.5
 	code.vegaprotocol.io/shared v0.0.0-20220614080106-5c97205b0d92
 	github.com/blang/semver v3.5.1+incompatible

--- a/go.mod
+++ b/go.mod
@@ -4,7 +4,7 @@ go 1.18
 
 require (
 	code.vegaprotocol.io/oracles-relay v0.0.0-20210201140234-f047e1bf6df3
-	code.vegaprotocol.io/protos v0.51.2-0.20220613103838-a73da28da1e1
+	code.vegaprotocol.io/protos v0.51.2-0.20220614140908-eb619915e744
 	code.vegaprotocol.io/quant v0.2.5
 	code.vegaprotocol.io/shared v0.0.0-20220614080106-5c97205b0d92
 	github.com/blang/semver v3.5.1+incompatible

--- a/go.mod
+++ b/go.mod
@@ -4,7 +4,7 @@ go 1.18
 
 require (
 	code.vegaprotocol.io/oracles-relay v0.0.0-20210201140234-f047e1bf6df3
-	code.vegaprotocol.io/protos v0.51.2-0.20220614140908-eb619915e744
+	code.vegaprotocol.io/protos v0.52.1-0.20220707090406-9ac0156ecee6
 	code.vegaprotocol.io/quant v0.2.5
 	code.vegaprotocol.io/shared v0.0.0-20220614080106-5c97205b0d92
 	github.com/blang/semver v3.5.1+incompatible

--- a/types/market.go
+++ b/types/market.go
@@ -106,6 +106,8 @@ const (
 	MarketTradingModeOpeningAuction MarketTradingMode = proto.Market_TRADING_MODE_OPENING_AUCTION
 	// Auction triggered by monitoring.
 	MarketTradingModeMonitoringAuction MarketTradingMode = proto.Market_TRADING_MODE_MONITORING_AUCTION
+	// No trading allowed.
+	MarketTradingModeNoTrading MarketTradingMode = proto.Market_TRADING_MODE_NO_TRADING
 )
 
 type MarketState = proto.Market_State

